url1a41b8e9f2_mr1azl_INF345_tgz-pJ8-TestParserJ8/src/xx/EG1.java

```diff
@@ -0,0 +1,396 @@
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+      term();
+    }
+  }
+
+  /**
+   * Method term.
+   * @throws ParseException
+   */
+  static final public void term() throws ParseException {
+    unary();
+    label_2:
+    while (true) {
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case MULTIPLY:
+      case DIVIDE:
+        ;
+        break;
+      default:
+        jj_la1[3] = jj_gen;
+        break label_2;
+      }
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case MULTIPLY:
+        jj_consume_token(MULTIPLY);
+        break;
+      case DIVIDE:
+        jj_consume_token(DIVIDE);
+        break;
+      default:
+        jj_la1[4] = jj_gen;
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+      unary();
+    }
+  }
+
+  /**
+   * Method unary.
+   * @throws ParseException
+   */
+  static final public void unary() throws ParseException {
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case MINUS:
+      jj_consume_token(MINUS);
+      element();
+      break;
+    case CONSTANT:
+    case 12:
+      element();
+      break;
+    default:
+      jj_la1[5] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+  }
+
+  /**
+   * Method element.
+   * @throws ParseException
+   */
+  static final public void element() throws ParseException {
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case CONSTANT:
+      jj_consume_token(CONSTANT);
+      break;
+    case 12:
+      jj_consume_token(12);
+      sum();
+      jj_consume_token(13);
+      break;
+    default:
+      jj_la1[6] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+  }
+
+  static private boolean jj_initialized_once = false;
+  /** Generated Token Manager. */
+  static public EG1TokenManager token_source;
+  static SimpleCharStream jj_input_stream;
+  /** Current token. */
+  static public Token token;
+  /** Next token. */
+  static public Token jj_nt;
+  static private int jj_ntk;
+  static private int jj_gen;
+  static final private int[] jj_la1 = new int[7];
+  static private int[] jj_la1_0;
+  static {
+      jj_la1_init_0();
+   }
+   private static void jj_la1_init_0() {
+      jj_la1_0 = new int[] {0x1a40,0x60,0x60,0x180,0x180,0x1240,0x1200,};
+   }
+
+  /** Constructor with InputStream. * @param stream java.io.InputStream
+   */
+  public EG1(java.io.InputStream stream) {
+     this(stream, null);
+  }
+  /** Constructor with InputStream and supplied encoding * @param stream java.io.InputStream
+   * @param encoding String
+   */
+  public EG1(java.io.InputStream stream, String encoding) {
+    if (jj_initialized_once) {
+      System.out.println("ERROR: Second call to constructor of static parser.  ");
+      System.out.println("       You must either use ReInit() or set the JavaCC option STATIC to false");
+      System.out.println("       during parser generation.");
+      throw new Error();
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 210:215, 245:250, 274:279.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.